src/gameRoom/state.js

```diff
@@ -1,7 +1,7 @@
 module.exports = {
   name: 'gameRoom',
   url: '/room/{gameId}',
-  onEnter: function($state, game, name) {
+  controller: function($state, game, name) {
```

Comment:
Hmm I think there is no other option.
It cant bu used on onEnter, cos it will do not replace URL

If you have other ideas please create PR